_utils/firefox_env_handler.py

```diff
@@ -0,0 +1,167 @@
+        pass
+
+    @staticmethod
+    def get_os():
+        """
+        Do our best to determine the user's current operating system.
+        """
+        system = platform.system().lower()
+        return re.split('[-_]', system, maxsplit=1).pop(0)
+
+    @classmethod
+    def is_linux(cls):
+        """
+        Am I running on Linux?
+        """
+        return cls.get_os() == cls.LINUX
+
+    @classmethod
+    def is_mac(cls):
+        """
+        Am I running on Mac/Darwin?
+        """
+        return cls.get_os() == cls.MAC
+
+    @classmethod
+    def is_other(cls):
+        """
+        I 'literally' have no idea who you are.
+        """
+        return not (cls.is_linux() or cls.is_mac() or cls.is_windows())
+
+    @classmethod
+    def is_windows(cls):
+        """
+        Am I running on Windows/Cygwin?
+        """
+        return cls.get_os() == cls.WINDOWS
+
+    @staticmethod
+    def banner(str, length=79, delimiter='='):
+        """
+        Throws up a debug header to make output subjectively "easier" to read
+        in stdout.
+        """
+        if length <= 0:
+            length = len(str)
+
+        divider = delimiter * length
+        output = '\n'.join(['', divider, str, divider])
+        print(output)
+
+    @staticmethod
+    def clean_folder(path, foot_gun=True):
+        """
+        Recursively delete the specified path.
+        """
+        if os.path.isdir(path):
+            print(('Deleting {0}'.format(path)))
+            shutil.rmtree(path)
+
+        else:
+            print(('{0} is not a directory'.format(path)))
+
+
+class IniHandler(FirefoxEnvHandler):
+    def __init__(self, ini_path=None):
+        """
+        Creates a new config parser object, and optionally loads a config file
+        if `ini_path` was specified.
+        """
+        self.config = configparser.SafeConfigParser(os.environ)
+
+        if ini_path is not None:
+            self.load_config(ini_path)
+
+    def load_config(self, ini_path):
+        """
+        Load an INI config based on the specified `ini_path`.
+        """
+        IniHandler.banner('LOADING {0}'.format(ini_path), 79, '-')
+
+        # Make sure the specified config file exists, fail hard if missing.
+        if not os.path.isfile(ini_path):
+            sys.exit('Config file not found: {0}'.format(ini_path))
+
+        self.config.read(ini_path)
+
+    def load_os_config(self, config_path):
+        """
+        Load an INI file based on the current operating system. This method
+        will attempt to load a "darwin.ini", "cygwin.ini", or "linux-gnu.ini"
+        file from the specified `config_path` based on the current OS.
+        """
+        os_config = os.path.join(config_path, IniHandler.get_os() + '.ini')
+        self.load_config(os_config)
+
+    def create_env_file(self, out_file='.env'):
+        """
+        Generate and save the output environment file so we can source it from
+        something like .bashrc or .bashprofile.
```

Comment:
nice! but yea you're right.  what's dirty is that we can only get away with files being generically named "run" with product tests because we know they'll always be embedded in a directory w/ the product name. we did this so jenkins can easily handle.  in the case of _utils it's more like a bucket-o-tools, so "run" looks like... run what?.
I recommend you rename: slave_refresh.py to: host_refresh.py
we'll add an arg parser to it so that it can be used more universally (to either refresh the jenkins slave or your laptop) as we were talking about.
then replace the _utils/run.sh  with _utils/jenkins_slave_refresh.sh
if the _utils dir gets too unwieldly, we can think of an alternate place for such scripts.